packages/theme-check-common/src/utils/warnings.ts

```diff
@@ -0,0 +1,21 @@
+import { getLocEnd, getLocStart, nodeAtPath } from '..';
+import { JSONNode } from '..';
+import { Context, SourceCodeType, ArrayNode } from '../types';
+
+export function reportWarning(
+  context: Context<SourceCodeType.LiquidHtml>,
+  offset: number,
+  ast: JSONNode,
+  ast_path: string[],
+  message: string,
+  fullHighlight: boolean = true,
+) {
+  const node = nodeAtPath(ast, ast_path)! as ArrayNode;
```

Comment:
I feel like this casting is dangerous given this method is extracted as a util 🤔?

Why not call context.report directly and do the casting when you need it instead? 

Would remove the need for the fullHighlight thing? 

reportWarning is pretty generic and it would feel weird to try to use that in a Liquid check, I'm not sure this deserves to be extracted?